### PR TITLE
Add a bunch or magic checks

### PR DESCRIPTION
**Pull request by gquintard**
this might be extremely OCD, hence the PR

### CODE REVIEW

File: lib/libvmod_unix/vmod_unix.c [hunk 1]
```diff
@@ -67,9 +67,12 @@ get_sp(VRT_CTX)
 {
 	struct sess *sp;
 
-	if (VALID_OBJ(ctx->req, REQ_MAGIC))
+	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
```
Comment (Dridi) [inline lib/libvmod_unix/vmod_unix.c]: CHECK_OBJ_ORNULL

Comment (Dridi) [reply lib/libvmod_unix/vmod_unix.c]: Or no check at all, since you check again in both `if` branches.

Comment (Dridi) [reply lib/libvmod_unix/vmod_unix.c]: Need coffee, wrong object, LTGM.

**Comment by bsdphk** [comment]
There is a macro for that: CHECK_OBJ_ORNULL(...)

**Comment by bsdphk** [comment]
Otherwise OK

File: bin/varnishd/cache/cache_vrt_var.c [hunk 2]
```diff
@@ -692,15 +700,26 @@ VRT_r_bereq_xid(VRT_CTX)
...
+	struct sess *sp;
+	struct http *http;
+
 	CHECK_OBJ_NOTNULL(ctx, VRT_CTX_MAGIC);
 
-	if (VALID_OBJ(ctx->req, REQ_MAGIC))
-		return (WS_Printf(ctx->req->http->ws, "%u",
-		    VXID(ctx->req->sp->vxid)));
+	if (ctx->req) {
+		CHECK_OBJ_NOTNULL(ctx->req, REQ_MAGIC);
```
[the remark this answers is not in the document]
Comment (gquintard) [reply bin/varnishd/cache/cache_vrt_var.c]: true, I did it for symmetry, but you are right, changing it

File: lib/libvmod_unix/vmod_unix.c [hunk 3]
```diff
@@ -67,9 +67,12 @@ get_sp(VRT_CTX)
...
+
+	if (ctx->req) {
+		CHECK_OBJ_NOTNULL(ctx->req, REQ_MAGIC);
```
[the remark this answers is not in the document]
Comment (gquintard) [reply lib/libvmod_unix/vmod_unix.c]: same

**Comment by gquintard** [comment]
pushed with one more tweak, and accounting for comments

**Pull request closed without merging**